README.md

```diff
@@ -1,2 +1,724 @@
-# purescript-affjax
-An asynchronous AJAX library built using Aff.
+# Module Documentation
+
+## Module Network.HTTP.Affjax
+
+#### `Ajax`
+
+``` purescript
+data Ajax :: !
+```
+
+The effect type for AJAX requests made with Affjax.
+
+#### `Affjax`
+
+``` purescript
+type Affjax e a = Aff (ajax :: Ajax | e) (AffjaxResponse a)
+```
+
+The type for Affjax requests.
+
+#### `AffjaxOptions`
+
+``` purescript
+data AffjaxOptions :: *
+```
+
+Options type for Affjax requests.
+
+#### `AffjaxResponse`
+
+``` purescript
+type AffjaxResponse a = { response :: a, headers :: [ResponseHeader], status :: StatusCode }
+```
+
+The type of records that will be received as an Affjax response.
+
+#### `url`
+
+``` purescript
+url :: Option AffjaxOptions String
+```
+
+Sets the URL for a request.
+
+#### `method`
+
+``` purescript
+method :: Option AffjaxOptions Method
+```
+
+Sets the HTTP method for a request.
+
+#### `content`
+
+``` purescript
+content :: Option AffjaxOptions RequestContent
+```
+
+Sets the content to send in a request.
+
+#### `headers`
+
+``` purescript
+headers :: Option AffjaxOptions [RequestHeader]
+```
+
+Sets the headers to send with a request.
+
+#### `username`
+
+``` purescript
+username :: Option AffjaxOptions String
+```
+
+Sets the HTTP auth username to send with a request.
+
+#### `password`
+
+``` purescript
+password :: Option AffjaxOptions String
+```
+
+Sets the HTTP auth password to send with a request.
+
+#### `affjax`
+
+``` purescript
+affjax :: forall e a. Responsable a -> Options AffjaxOptions -> Affjax e a
+```
+
+Runs a request.
+
+#### `affjax'`
+
+``` purescript
+affjax' :: forall e a. Responsable a -> Options AffjaxOptions -> (Error -> Eff (ajax :: Ajax | e) Unit) -> (AffjaxResponse a -> Eff (ajax :: Ajax | e) Unit) -> Eff (ajax :: Ajax | e) Unit
+```
+
+Runs a request directly in Eff.
+
+#### `get`
+
+``` purescript
+get :: forall e a. Responsable a -> String -> Affjax e a
```

Comment:
In all these methods, I'd put the URL first, because you are likely to make many requests to the same URL, whereas each request is likely to have different content. Also I'd use a `type URL = String` just for doc purposes.